#! /bin/sh /usr/share/dpatch/dpatch-run
# Description: add pkexec policy file
# Author: Marc Deslauriers <marc.deslauriers@canonical.com>

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' synaptic-0.75.2ubuntu4~/data/Makefile.am synaptic-0.75.2ubuntu4/data/Makefile.am
--- synaptic-0.75.2ubuntu4~/data/Makefile.am	2011-08-02 04:11:58.000000000 -0400
+++ synaptic-0.75.2ubuntu4/data/Makefile.am	2011-08-11 08:42:16.968978987 -0400
@@ -1,4 +1,5 @@
 @INTLTOOL_DESKTOP_RULE@
+@INTLTOOL_POLICY_RULE@
 
 desktopdir       = $(datadir)/applications
 desktop_in_files = synaptic.desktop.in
@@ -8,7 +9,12 @@
 kdeapp_in_files = synaptic-kde.desktop.in
 kdeapp_DATA     = $(kdeapp_in_files:.desktop.in=.desktop)
 
+polkit_policydir = $(datadir)/polkit-1/actions
+dist_polkit_policy_in_files = com.ubuntu.pkexec.synaptic.policy.in
+dist_polkit_policy_DATA = $(dist_polkit_policy_in_files:.policy.in=.policy)
+
 EXTRA_DIST = $(desktop_in_files) \
 		$(desktop_DATA) \
 		$(kdeapp_in_files) \
-		$(kdeapp_DATA) 
\ No newline at end of file
+		$(kdeapp_DATA) \
+		$(dist_polkit_policy_in_files)
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' synaptic-0.75.2ubuntu4~/data/com.ubuntu.pkecec.synaptic.policy.in synaptic-0.75.2ubuntu4/data/com.ubuntu.pkecec.synaptic.policy.in
--- synaptic-0.75.2ubuntu4~/data/com.ubuntu.pkecec.synaptic.policy.in	1969-12-31 19:00:00.000000000 -0500
+++ synaptic-0.75.2ubuntu4/data/com.ubuntu.pkecec.synaptic.policy.in	2011-08-11 08:40:45.478979011 -0400
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE policyconfig PUBLIC
+ "-//freedesktop//DTD PolicyKit Policy Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/PolicyKit/1/policyconfig.dtd">
+<policyconfig>
+
+  <action id="com.ubuntu.pkexec.synaptic">
+    <_message>Authentication is required to run the Synaptic Package Manager</_message>
+    <icon_name>synaptic</icon_name>
+    <defaults>
+      <allow_any>auth_admin</allow_any>
+      <allow_inactive>auth_admin</allow_inactive>
+      <allow_active>auth_admin</allow_active>
+    </defaults>
+    <annotate key="org.freedesktop.policykit.exec.path">/usr/sbin/synaptic</annotate>
+    <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
+  </action>
+
+</policyconfig>
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' synaptic-0.75.2ubuntu4~/data/com.ubuntu.pkexec.synaptic.policy synaptic-0.75.2ubuntu4/data/com.ubuntu.pkexec.synaptic.policy
--- synaptic-0.75.2ubuntu4~/data/com.ubuntu.pkexec.synaptic.policy	1969-12-31 19:00:00.000000000 -0500
+++ synaptic-0.75.2ubuntu4/data/com.ubuntu.pkexec.synaptic.policy	2011-08-11 08:40:44.388979011 -0400
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE policyconfig PUBLIC
+ "-//freedesktop//DTD PolicyKit Policy Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/PolicyKit/1/policyconfig.dtd">
+<policyconfig>
+
+  <action id="com.ubuntu.pkexec.synaptic">
+    <message>Authentication is required to run the Synaptic Package Manager</message>
+    <icon_name>synaptic</icon_name>
+    <defaults>
+      <allow_any>auth_admin</allow_any>
+      <allow_inactive>auth_admin</allow_inactive>
+      <allow_active>auth_admin</allow_active>
+    </defaults>
+    <annotate key="org.freedesktop.policykit.exec.path">/usr/sbin/synaptic</annotate>
+    <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
+  </action>
+
+</policyconfig>
